src/main/java/seedu/address/logic/commands/UnmatchCommand.java

```diff
@@ -0,0 +1,146 @@
+import seedu.address.model.skill.Skill;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Unmatch a contact from their job.
+ */
+public class UnmatchCommand extends Command {
+    public static final int CONTACT_INDEX_POS = 0;
+    public static final int JOB_INDEX_POS = 1;
+    public static final String COMMAND_WORD = "unmatch";
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Unmatch a contact to a job\nParameters: <CONTACT_INDEX> <JOB_INDEX>\nExample: "
+                    + COMMAND_WORD + " 2 1";
+
+    public static final String MESSAGE_UNMATCH_SUCCESS = "Unmatched Contact: %1$s with Job: %2$s";
+    public static final String MESSAGE_ALREADY_UNMATCHED = "Contact already unmatched with this job!";
+
+    private final Index contactIndex;
+    private final Index jobIndex;
+
+    /**
+     * @param contactIndex Index of the contact in the filtered person list to unmatch
+     * @param jobIndex Index of the job in the filtered contact list to unmatch
+     */
+    public UnmatchCommand(Index contactIndex, Index jobIndex) {
+        requireNonNull(contactIndex);
+        requireNonNull(jobIndex);
+
+        this.contactIndex = contactIndex;
+        this.jobIndex = jobIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Person> lastShownPeronList = model.getFilteredPersonList();
+        if (contactIndex.getZeroBased() >= lastShownPeronList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+        }
+
+        List<Job> lastShownJobList = model.getFilteredJobList();
+        if (jobIndex.getZeroBased() >= lastShownJobList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+        }
+
+        Person contactToUnmatch = lastShownPeronList.get(contactIndex.getZeroBased());
+        Job jobToUnmatch = lastShownJobList.get(jobIndex.getZeroBased());
+
+        assert contactToUnmatch != null;
+        assert jobToUnmatch != null;
+
+        // TODO: Assume contact name is used to preserve uniqueness for now
+        final String contactIdentifier = contactToUnmatch.getIdentifier();
+        final String jobIdentifier = jobToUnmatch.getIdentifier();
+
+        // Check if the contact is already unmatched from the job
+        boolean hasContactMatchedJob = contactToUnmatch.hasMatched(jobIdentifier);
+        boolean hasJobMatchedContact = jobToUnmatch.hasMatched(contactIdentifier);
+
+        if (!hasContactMatchedJob && !hasJobMatchedContact) {
+            throw new CommandException(MESSAGE_ALREADY_UNMATCHED);
+        }
+
+        assert hasContactMatchedJob == hasJobMatchedContact;
+        // Unmatch the contact from the job by removing the associations
+        Person unmatchedContact = unmatchContactFromJob(contactToUnmatch);
+        Job unmatchedJob = unmatchJobFromContact(jobToUnmatch, contactIdentifier);
```

Comment:
The naming of these functions seems a bit unclear, for the first line, if you are trying to remove the reference of a job in a contact and return said contact, shouldn't the name be `unmatchJobFromContact`?

Perhaps a clearer method name would be `removeJobFromContact`? Not very sure.

Reply:
@Green-Tea-123 perhaps you could share your take on this?

Reply:
I agree

Reply:
@Green-Tea-123 thanks for the input! May I just check out of the 3 options I listed, which one do you think is the best?

Reply:
I'd say the second as its meaning is clear and job and contact preserve their subject-object role.